### DISCUSSION TITLE
Problem: Long Videos Are Truncated or Incomplete

### DISCUSSION THREAD

**Discussion by alidubais** [aclap-dev/video-downloadhelper#1146] [Help and Q&A]
### Your question

**I am new to VDH, but already purchased a license. 
I am facing a problem when trying to download long lectures approx. 2.5 hours long each. VDH shows a short duration of approximately 30 minutes only. It does download the 30 minutes fine, but how do I download the whole video lecture?**

**This is happening on Chrome, running on Windows 11 Pro.**

### Addon & CoApp version and details

version: 9.1.0.43
target: google
channel: stable
lang: en-US
coapp: {"found":true,"new_version":false,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19"}
license: {"accepted":true,"email":"<removed>","key":"81d56cf48f8d4890"}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/127.0.0.0 Safari/537.36
{
    "valid_license_message_has_been_displayed": true,
    "http_media_download_strategy": "coapp",
    "used_history_button": true,
    "last_download_directory": {
        "__serializer_tag": "option_some",
        "__serializer_value": {
            "__serializer_tag": "primitive",
            "__serializer_value": "C:\\Users\\alidu\\Downloads"
        }
    },
    "never_show_no_incognito_msg_again": true,
    "view_options": {
        "all_tabs": false,
        "hide_downloaded": false,
        "low_quality": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "sort_by_status": true,
        "sort_reverse": false
    },
    "use_sidebar": true,
    "open_count_store": 1,
    "successfull_dl": 2,
    "dable_attempts": {
        "__serializer_tag": "map",
        "__serializer_value": [
            {
                "__serializer_tag": "array",
                "__serializer_value": [
                    {
                        "__serializer_tag": "primitive",
                        "__serializer_value": "efficaciousleadership.com"
                    },
                    {
                        "__serializer_tag": "array",
                        "__serializer_value": [
                            {
                                "__serializer_tag": "primitive",
                                "__serializer_value": 5
                            },
                            {
                                "__serializer_tag": "primitive",
                                "__serializer_value": 2
                            }
                        ]
                    }
                ]
            }
        ]
    },
    "dable_attempts_last": 1723313989654
}

**Comment by alidubais** [comment]
![VDH Long Video Problem](https://github.com/user-attachments/assets/eb78d575-de5a-47af-80e8-791d7a0c8c05)

**Reply by alidubais** [reply]
**This is an example video.**

**Reply by Mattmjs** [reply]
It should download the whole video, it might be a display issue for video over an hour long with your image showing it 
as MPD. This indicates it's a dash stream video. Don't worry about what it shows you, just let it download to completion
and it should be the full duration.

**Reply by Mattmjs** [reply]
Further to that, it looks to be on Vimeo hosted videos,  #123 & #172

**Comment by woonwg** [comment]
same here, any solution?

**Comment by GrampaWildWilly** [comment]
It would seem like telling us where the problematic video is might get you some meaningful help.  Some URLs would be nice.  I'm not seeing the problem so I'd need some examples from you.

**Comment by alidubais** [comment]
I would have shared URLs to demonstrate the problem but the videos are
actually private/subscription based.

**Reply by GrampaWildWilly** [reply]
OK.  As an alternative to that, find the Hit Details for one of the items in question & post the `#0 main url` from those details.  Sometimes, that can be useful.  Sometimes.  No guarantees.  But it's the best we can do given that this content is behind a paywall.

**Reply by GrampaWildWilly** [reply]
Also, if you're going to post here via E-mail, do please do the polite thing & delete all quoted material before you send.  Quoting previous posts, which I assume you receive via E-mail, is entirely unnecessary.  Those previous posts are already here on GitHub.  So you're just posting wasted duplication.  Don't do that.

**Reply by alidubais** [reply]
Sorry, I am new here. Didn't realize it is going to quote EVERYTHING.

**Reply by alidubais** [reply]
> OK. As an alternative to that, find the Hit Details for one of the items in question & post the `#0 main url` from those details. Sometimes, that can be useful. Sometimes. No guarantees. But it's the best we can do given that this content is behind a paywall.

Not sure what you meant by Hit Details, but the website in question is https://efficaciousleadership.com/

**Reply by GrampaWildWilly** [reply]
For E-mail, most E-mail clients can be kind of sneaky about quoting.  People are usually not 100% aware that quoting is going on.  For exchanges of E-mails, I suppose it can be helpful.  But for posting in a forum, it's just bloat.  You would realize that any time you actually log on here & view the discussion in your browser.  Aside from the quoting, there is a shortcoming of sending E-mails to GitHub in particular.  Maybe it works on other forums.  It used to work on our Google Groups forum.  But GitHub does not accept attachments on E-mailed posts.  You can attach the items at your end.  When you send, it looks like it worked just fine.  But when the E-mail reaches GitHub, the web site strips off the attachments.  If you want to post attachments, the only way to do it is by visiting GitHub in your browser & using the controls underneath the text entry box for a post.

To find Hit Details . . .

![#01](https://github.com/user-attachments/assets/e8d5d736-6958-46d4-a0de-ea01dbe48815)
![#02](https://github.com/user-attachments/assets/ca911367-24da-4bf2-b34d-014dae2f999d)

I'm not going to post an example here.  It's a long page of rather tedious details.  But those details can sometimes be useful.  Since we're not actually interested in the Hit Details of that particular YouTube page, I'm not going to post them.  But in general, when somebody asks for the Hit Details, you just do this:

- Ctrl+a to highlight the entire page of Hit Details
- Ctrl+Insert to copy the Hit Details to your clipboard
- Switch tasks to a post here on GitHub
- Shift+Insert to paste the Hit Details into a post here

That particular sequence is required.  Anything else you might try interferes with the proper transfer of the Hit Details into a post.

Now, to the real topic at hand.  I opened your URL in a new Firefox window.  Firefox offered me the usual dialog asking what to do with it.  It offered to open it in an application of my choosing, which I didn't want to do.  It also offered to download it for me.  But the download would have just downloaded the file.  The file is a DASH master manifest.  I know that partly from the .mpd that appears in the middle of your URL.  I know it also because of what comes next.  I did not accept either choice Firefox offered me.  I simply dismissed the dialog box.  I now had a completely empty browser page.  But VDH had recognized something.  It claimed to be a video of duration 31:20 & resolution 720p.  So I told VDH to download the item.  After a time, VDH stopped showing a download completed percentage & just went to a timer telling me how long it had been downloading.  This is typical of livestreams.  But this is most assuredly not a livestream.  I guess however the web site was presenting the video, VDH interpreted it as a livestream.

This download proceeded at about 350,000 bytes per second.  That is horrifically slow.  I was tempted to just cancel the download on the assumption that it would eventually complete.  But that would not be a proper test.  You are saying VDH refuses to download the whole thing.  So I put my patience cap back on & let it run.  This is what I eventually got.

![#03](https://github.com/user-attachments/assets/37da69ac-6b4d-493b-8a28-ddb100349f93)
![#04](https://github.com/user-attachments/assets/12018e94-4e96-4ac9-aee6-cb95d19ad19f)

I didn't sit & watch it, just sampled it at 1-minute intervals.  There was both video & audio at each sample point.  I paid special attention to look at the last 10 seconds of the video.  It did a gentle fade to black at the end.  If I didn't have the whole thing, I wouldn't expect such an ending.  An incomplete video would just abruptly end seemingly in the middle of a frame.  But this one faded.  There's no end credits so there is still a tiny doubt in my mind that I have the whole thing.  But I'm pretty sure it's all of it.  You would know what the real duration is.

Notice how despite my not being authorized on the site, the URL you posted allowed me to download the video anyway.  We got lucky here.  This is why I ask for Hit Details as a fallback approach to diagnosing a problem.  This does not always work.  But like I say, we got lucky here.

My setup is Windows 11 Pro 64-bit version 23H2 build 22631.4037, Firefox 129.0.2 64-bit, licensed VDH 9.1.0.41 beta, CoApp 2.0.20.  The main difference between your setup & mine is the browser.  I'm on Firefox while you are on Chrome.  Being on Firefox gives me access to the VDH beta.  There used to also be VDH betas for Chrome & Edge but Paul has quit publishing those.  I'm sure it's a lot of work to put up 3 betas as opposed to only 1.  But it kind of leaves you hanging until Paul publishes a general availability VDH for your browser.

My conclusion is that the beta contains some bug fix that solves your problem.  Depending on your level of motivation, you could convert to using Firefox.  Depending on an even higher level of motivation, you could probably download your content without using any browser add-on.  I can not verify this (but you could, being an authorized user of the web site) but I would assume that in the Network Monitor in the browser Developer Tools, you could find the DASH master manifest for any of your lectures.  It would then be a piece of cake to download anything you wanted with ffmpeg.

There is another issue that a lot of people are encountering on education web sites.  The authors of the lessons are protecting their content with DRM.  This seems to be haphazard, with some authors doing it & some not, some authors protecting parts of courses while leaving other parts unprotected.  Only unprotected content can be downloaded.  That applies to both VDH & ffmpeg.  Consider yourself lucky that there doesn't appear to be any DRM protection in place on your lectures.  Well, I shouldn't jump to that conclusion.  This lecture isn't protected.  Each lecture would be a separate case for you to make that determination.

**Reply by alidubais** [reply]
Many many thanks, you really did a lot of work to reach into that conclusion - and you are absolutely right: you got the whole thing. 
I have actually tried your suggestion on Chrome, and can confirm that VDH ended up downloading the whole lecture. I don't mind if VDH was not showing the true duration of the video, as am able to download the whole thing for offline viewing anyway.

**Reply by JoeH-tw** [reply]
i know the topic was posted for long time ago. 
however, i can't understand one of sequence in answer.
could anyone share where to past Hit Details ?

- Ctrl+a to highlight the entire page of Hit Details
- Ctrl+Insert to copy the Hit Details to your clipboard
- Switch tasks to a post **here** on GitHub**
- Shift+Insert to paste the Hit Details into a post here

**Reply by Mattmjs** [reply]
You paste it as a reply here, or paste it in a new discussion of your own detailing your problem.

**Comment by garry-ut99** [comment]
Similar problem for me recently:
- https://github.com/aclap-dev/video-downloadhelper/discussions/2969#discussioncomment-15385454
